scene/gui/spin_box.cpp

```diff
@@ -63,20 +63,33 @@ void SpinBox::_update_text(bool p_only_update_if_value_changed) {
 			value += " " + suffix;
 		}
 	}
+
+	String text = line_edit->get_text().replace(",", ".");
+
+	if (!accepted && update_on_text_changed && !text.contains(".")) {
```

Comment:
It would be better if `text` was inside the condition. It's irrelevant unless `!accepted && update_on_text_changed` are true.